File: VehicleRouting/sample/java/VehicleRouting.java
import java.io.*;
import java.security.*;
import java.util.*;

public class VehicleRouting {

    public static void main (String[] args) {
        try {
            Scanner sc = new Scanner(System.in);
            int N = sc.nextInt();
            int M = sc.nextInt();
            int depotX = sc.nextInt();
            int depotY = sc.nextInt();
            int posX[] = new int[N];
            int posY[] = new int[N];
            for (int i = 0; i < N; i++) {
                posX[i] = sc.nextInt();
                posY[i] = sc.nextInt();
            }
            int cap[] = new int[M];
            int speed[] = new int[M];
            for (int i = 0; i < M; i++) {
                cap[i] = sc.nextInt();
                speed[i] = sc.nextInt();
            }
            int ans_size[] = new int[M];
            int ans[][] = new int[M][N];
            for (int i = 0; i < N; i++) {
                ans[i % M][ans_size[i % M]++] = i;
            }
            for (int i = 0; i < M; ++i) {
                System.out.print(ans_size[i]);
                for (int j = 0; j < ans_size[i]; j++) {
                    System.out.print(" " + ans[i][j]);
                }
                System.out.println();
            }
        }
        catch (Exception e) {

        }
    }

}

File: VehicleRouting/sample/python/VehicleRouting.py
import sys

N,M = map(int,raw_input().split())
depotX,depotY = map(int,raw_input().split())

posX = [];
posY = [];
for i in range(N):
	p = map(int,raw_input().split())
	posX.append(p[0])
	posY.append(p[1])

cap = []
speed = []
for i in range(M):
	p = map(int,raw_input().split())
	cap.append(p[0])
	speed.append(p[1])

ans = [[0 for i in range(0)] for j in range(M)]

for i in range(N):
	ans[i % M].append(i)

for i in range(M):
	print len(ans[i])
	for j in ans[i]:
		print j

sys.stdout.flush()
